smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/CommandGenerator.java

```diff
@@ -474,7 +484,29 @@ private void addCommandSpecificPlugins() {
                 }
                 writer.pushState();
                 writer.putContext(symbolMap);
-                writer.write("$pluginFn:T(config" + additionalParamsString + "),");
+                writer.openBlock("$pluginFn:T(config", "),", () -> {
+                    List<String> additionalParameters = CodegenUtils.getFunctionParametersList(paramsMap);
+                    Map<String, CommandWriterConsumer> clientAddParamsWriterConsumers =
+                        plugin.getOperationAddParamsWriterConsumers();
+                    if (additionalParameters.isEmpty() && clientAddParamsWriterConsumers.isEmpty()) {
+                        return;
+                    }
+                    writer.writeInline(", { ");
+                    writer.writeInline(String.join(", ", additionalParameters));
```

Comment:
nit: You could use templating and add additional params to the context
```
writer.writeInline(", {${#params}${value:L}${^key.last}, ${/key.last}${/params}")
```

Reply:
I don't understand this syntax at a glance. Are there docs for this?

Reply:
https://smithy.io/2.0/guides/building-codegen/generating-code.html